usefulUtilities/statusIndicator/hifiScript/statusIndicator.js

```diff
@@ -29,35 +27,46 @@
         OVERLAY_RECTANGLE_TOP_PADDING = 12,
         OVERLAY_RECTANGLE_WIDTH_HEIGHT = 20,
         AVAILABLE_COLOR = { red: 0, green: 144, blue: 54 },
-        NOT_AVAILABLE_COLOR = { red: 255, green: 0, blue: 26 };
-    function drawStatusOverlays() {
-        var windowWidth = Window.innerWidth;
-
-        desktopOverlay = Overlays.addOverlay("text", {
-            text: isAvailable ? OVERLAY_AVAILABLE_TEXT : OVERLAY_NOT_AVAILABLE_TEXT,
+        NOT_AVAILABLE_COLOR = { red: 255, green: 0, blue: 26 },
+        OTHER_COLOR = { red: 255, green: 237, blue: 0 },
+        TEXT_OVERLAY_PROPS = {
+            width: OVERLAY_RECTANGLE_WIDTH_HEIGHT,
+            height: OVERLAY_RECTANGLE_WIDTH_HEIGHT,
+            y: OVERLAY_FONT_SIZE + OVERLAY_RECTANGLE_TOP_PADDING,
+            color: { red: 255, green: 255, blue: 255 },
+            alpha: 1.0,
+            visible: true
+        },
+        RECTANGLE_OVERLAY_PROPS = {
             width: OVERLAY_WIDTH,
             height: OVERLAY_FONT_SIZE * 2.2,
-            x: windowWidth - OVERLAY_WIDTH - OVERLAY_WINDOW_RIGHT_PADDING,
             y: OVERLAY_FONT_SIZE,
             lineHeight: OVERLAY_FONT_SIZE,
             margin: OVERLAY_FONT_SIZE / 2,
             font: { size: OVERLAY_FONT_SIZE },
-            color: { red: 255, green: 255, blue: 255 },
             backgroundColor: { red: 17, green: 17, blue: 17 },
             alpha: 1.0,
             backgroundAlpha: 0.7,
             visible: true
-        });
+        };
+        
+    function drawStatusOverlays() {
+        var windowWidth = Window.innerWidth;
 
-        rectangleOverlay = Overlays.addOverlay("rectangle", {
-            width: OVERLAY_RECTANGLE_WIDTH_HEIGHT,
-            height: OVERLAY_RECTANGLE_WIDTH_HEIGHT,
-            x: windowWidth - OVERLAY_WIDTH + OVERLAY_RECTANGLE_LEFT_PADDING - OVERLAY_WINDOW_RIGHT_PADDING,
-            y: OVERLAY_FONT_SIZE + OVERLAY_RECTANGLE_TOP_PADDING,
-            color: isAvailable ? AVAILABLE_COLOR : NOT_AVAILABLE_COLOR,
-            alpha: 1.0,
-            visible: true
-        });
+        var desktopOverlayProps = RECTANGLE_OVERLAY_PROPS;
+        desktopOverlayProps.x = windowWidth - OVERLAY_WIDTH - OVERLAY_WINDOW_RIGHT_PADDING;
+        desktopOverlay = Overlays.addOverlay("text", desktopOverlayProps);
+
+        var rectangleOverlayProps = TEXT_OVERLAY_PROPS;
+        rectangleOverlayProps.x = windowWidth - OVERLAY_WIDTH + OVERLAY_RECTANGLE_LEFT_PADDING - OVERLAY_WINDOW_RIGHT_PADDING;
+        if (currentStatus === "available") {
+            rectangleOverlayProps.color = AVAILABLE_COLOR;
+        } else if (currentStatus === "busy") {
+            rectangleOverlayProps.color = NOT_AVAILABLE_COLOR;
+        } else {
+            rectangleOverlayProps.color = OTHER_COLOR;
+        }
+        rectangleOverlay = Overlays.addOverlay("rectangle", rectangleOverlayProps);
     }
 
 
```

Comment:
I'm unable to comment below on lines 77 or 81. When the overlays are unavailable: rectangleOverlay = false; desktopOverlay = false

What do you think?

Reply:
ah yes, for consistency, good call